libraries/abstractions/common_io/test/iot_test_common_io.c

```diff
@@ -88,4 +88,116 @@ TEST_GROUP_RUNNER( Common_IO )
             RUN_TEST_GROUP( TEST_IOT_SPI );
         }
     #endif
+
+    #ifdef IOT_TEST_COMMON_IO_WATCHDOG_SUPPORTED
+        for( i = 0; i < IOT_TEST_COMMON_IO_WATCHDOG_SUPPORTED; i++ )
+        {
+            SET_TEST_IOT_WATCHDOG_CONFIG( i );
+            RUN_TEST_GROUP( TEST_IOT_WATCHDOG );
+        }
+    #endif
+
+    #ifdef IOT_TEST_COMMON_IO_RTC_SUPPORTED
+        for( i = 0; i < IOT_TEST_COMMON_IO_RTC_SUPPORTED; i++ )
+        {
+            SET_TEST_IOT_RTC_CONFIG( i );
+            RUN_TEST_GROUP( TEST_IOT_RTC );
+        }
+    #endif
+
+    #ifdef IOT_TEST_COMMON_IO_RESET_SUPPORTED
+        for( i = 0; i < IOT_TEST_COMMON_IO_RESET_SUPPORTED; i++ )
+        {
+            SET_TEST_IOT_RESET_CONFIG( i );
+            RUN_TEST_GROUP( TEST_IOT_RESET );
+        }
+    #endif
+
+    #ifdef IOT_TEST_COMMON_IO_PERFCOUNTER_SUPPORTED
```

Comment:
Same as for Timer and Flash. Should this test be located before the GPIO tests due to the potential dependency?

Reply:
Thanks for catching this as well, will apply this feedback